### DISCUSSION TITLE
getSortedPostsData "is not a function"

### DISCUSSION THREAD

**Discussion by drobert-bfm** [vercel/next.js#40054] [Help] [answered]
Getting a TypeError when going through the tutorial, section "Pre-rendering and Data Fetching", part 7:

TypeError: (0 , _lib_posts__WEBPACK_IMPORTED_MODULE_3__.getSortedPostsData) is not a function

Code is exactly as specified, copy/pasted from tutorial.

**Comment by drobert-bfm** [comment, marked as answer]
If anyone else has this issue: add a carriage return at the end of posts.js.

**Reply by windisch-paul** [reply]
That fixed it indeed! Can you elaborate on why that happens and how you debugged it?

**Comment by nirajanach** [comment]
If you still have the same problem then delete the posts.js file and create it again and paste the code( make sure there is carriage return ( new line) at the end of the code. It worked
